src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -3327,6 +3444,7 @@ bool BlockchainLMDB::get_output_distribution(uint64_t amount, uint64_t from_heig
   MDB_val_set(k, amount);
   MDB_val v;
   MDB_cursor_op op = MDB_SET;
+  base = 0;
```

Comment:
Well, it's used by an external class. I don't see where you're going here.

Reply:
Why would this function still be exposed for use externally, given its performance?

Reply:
How else would it be used ? Or are you saying non rct outs should keep using a triangular distribution, and not use segregation ? If so, I suppose they're rare nowadays so that could fly...

Reply:
This is actually not very slow when used for a small number of outputs. What's slow is (1) iterating through all the huge amounts of dust outputs and (2) rct outs, of which there are a lot. (1) is irelevant if you're using a trusted daemon, and (2) is mooted by this PR.